librecad/src/lib/generators/lc_makercamsvg.cpp

```diff
@@ -543,6 +550,193 @@ void LC_MakerCamSVG::writeEllipse(RS_Ellipse* ellipse) {
+
+        writeQuadraticBeziers(spline->getControlPoints(), spline->isClosed());
+    }
+    else if (spline->getDegree() == 3) {
+        RS_DEBUG->print("RS_MakerCamSVG::writeSpline: Writing piecewise cubic spline as 'path' with cubic bézier segments");
+
+        writeCubicBeziers(spline->getControlPoints(), spline->isClosed());
+    }
+    else {
+        RS_DEBUG->print(RS_Debug::D_NOTICE,
+                        "RS_MakerCamSVG::writePiecewiseCubicSpline: Splines with degree '%d' not implemented",
+                        (int)spline->getDegree());
+    }
+}
+
+void LC_MakerCamSVG::writeSplinepoints(LC_SplinePoints* splinepoints) {
+
+    RS_DEBUG->print("RS_MakerCamSVG::writeSplinepoints: Writing piecewise quadratic spline as 'path' with quadratic bézier segments");
+
+    writeQuadraticBeziers(splinepoints->getControlPoints(), splinepoints->isClosed());
+}
+
+void LC_MakerCamSVG::writeCubicBeziers(const std::vector<RS_Vector> &control_points, bool is_closed) {
+
+    std::vector<RS_Vector> bezier_points = calcCubicBezierPoints(control_points, is_closed);
+
+    std::string path = svgPathMoveTo(convertToSvg(bezier_points[0]));
+
+    int bezier_points_size = bezier_points.size();
+
+    int bezier_count = ((bezier_points_size - 1) / 3);
+
+    for (int i = 0; i < bezier_count; i++) {
+        path += svgPathCurveTo(convertToSvg(bezier_points[3 * (i + 1)]), convertToSvg(bezier_points[3 * (i + 1) - 2]), convertToSvg(bezier_points[3 * (i + 1) - 1]));
+    }
+
+    xmlWriter->addElement("path", NAMESPACE_URI_SVG);
+
+    xmlWriter->addAttribute("d", path);
+
+    xmlWriter->closeElement();
+}
+
+void LC_MakerCamSVG::writeQuadraticBeziers(const std::vector<RS_Vector> &control_points, bool is_closed) {
+
+    std::vector<RS_Vector> bezier_points = calcQuadraticBezierPoints(control_points, is_closed);
+
+    std::string path = svgPathMoveTo(convertToSvg(bezier_points[0]));
+
+    int bezier_points_size = bezier_points.size();
+
+    int bezier_count = ((bezier_points_size - 1) / 2);
+
+    for (int i = 0; i < bezier_count; i++) {
+        path += svgPathQuadraticCurveTo(convertToSvg(bezier_points[2 * (i + 1)]), convertToSvg(bezier_points[2 * (i + 1) - 1]));
+    }
+
+    xmlWriter->addElement("path", NAMESPACE_URI_SVG);
+
+    xmlWriter->addAttribute("d", path);
+
+    xmlWriter->closeElement();
+}
+
+std::vector<RS_Vector> LC_MakerCamSVG::calcCubicBezierPoints(const std::vector<RS_Vector> &control_points, bool is_closed) {
+
+    std::vector<RS_Vector> bezier_points;
+
+    int control_points_size = control_points.size();
+
+    int bezier_points_size;
+
+    if (is_closed) {
+
+        for (int i = 0; i < (control_points_size - 1); i++) {
+            bezier_points.push_back(control_points[i]);
+
+            bezier_points.push_back((control_points[i] * 2.0 + control_points[i + 1]) / 3.0);
+            bezier_points.push_back((control_points[i] + control_points[i + 1] * 2.0) / 3.0);
+        }
+
```

Comment:
I am not deep into this.

Is the linear interpolation form, (p0, (2/3)*p0 + (1/3)*p1, (1/3)*p0 + (2/3)*p1, p1), leading to quadratic spline points (lc_splinepoints)?

If not, we may want to consider a second order spline interpolation.